### PR TITLE
Data Access With ADO.NET In Micro Focus Visual COBOL

### LINKED ISSUE #2946: Data Access With ADO.NET In Micro Focus Visual COBOL
### Proposed title of article
	
Data Access With ADO.NET In Micro Focus Visual COBOL

### Introduction paragraph (2-3 paragraphs):
ADO is the acronym for ActiveX Data Object and is based on Microsoft technology. Hence, ADO.NET which is the next evolutionary step in data access technology. Employing the aforementioned ADO technology, ADO.NET expands this by incorporating XML into a standard model to not only the relational data models but also the text based XML data.

This will be a multi-part article dealing with data access in the Microsoft .NET arena, using ADO. We will begin with a bit of history of ADO, what it is, why it's so important and finally how to utilize it in a COBOL environment. This article will provide a brief history of the data access as well as provide the basic flow for using ADO to access the data. 

### Key takeaways: 
By the end of this article, the reader will understand:
1. what ADO and ADO.NET are.
2. Basic ADO flow.
- preparation.
- Create a connection to the data object.
- Create a command for the data object to act upon.
- Read the data returned by the data object.
3. what wrap up is.

### References: 
N/A

### PR DESCRIPTION
**Pull request by Amoh-writer**
Data Access With ADO.NET In Micro Focus Visual COBOL

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2946

### CODE REVIEW

**Comment by Amoh-writer** [comment]
Hello @hectorkambow what does the `test` mean?

**Comment by jerimkaura** [comment]
> Hello @hectorkambow what does the `test` mean?

I guess he was testing if comments work, GitHub was down for a short moment. :smile:

**Comment by WanjaMIKE** [comment]
@Amoh-writer 
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @Amoh-writer,

Welcome to EngEd community!

Before we get started with the review, could you please add authors `index.md` and `avatar.jpeg` under `/content/authors/[ENTER-YOUR-NAME-IN-LOWERCASE]` (Please refer other files to get an idea)

And, also could you modify the article name (article directory) from `content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol` to `content/articles/data-access-with-ado-net-in-micro-focus-visual-cobol`.

Ping me when done.

Feel free to reach out to me here or Slack, in case of any questions.

Thanks.

**Comment by Amoh-writer** [comment]
@srishilesh Hope you are doing great. Wish to inquire how I will change the article directory. Thank you

**Comment by srishilesh** [comment]
@Amoh-writer 

To change the article directory, just rename the folder to `content/articles/data-access-with-ado-net-in-micro-focus-visual-cobol`.

**Comment by hectorkambow** [comment]
@Amoh-writer 

[Data Access With ADO.NET In Micro Focus Visual COBOL #3260.pdf](https://github.com/section-engineering-education/engineering-education/files/6994718/Data.Access.With.ADO.NET.In.Micro.Focus.Visual.COBOL.3260.pdf)

👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by srishilesh** [comment]
@Amoh-writer 

Could you rework based on these plagiarism checks, along with the existing suggested changes? And, ping me when you're done.

If you're still facing issues with renaming the directory, I'll take care of it. Work on the remaining suggestions.

Thanks.

**Comment by Amoh-writer** [comment]
@srishilesh done correcting

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+- [Preparation](#preparation)
+- [Create a Connection](#create-a-connection)
+- [Create a Command](#create-a-command)
+- [Read the Data](#read-the-data)
+- [Housekeeping](#housekeeping)
+- [Other Data Providers](#other-data-providers)
+- [Wrap-Up](#wrap-up)
+
+### Basic ADO Flow
+What do I do now that Microsoft has provided me with this one-of-a-kind library of data access procedures? How can I get started? What do I look for when I am looking for a place to stay? We will start by looking at `SQL Server's namespace`, which governs data access. At the end of this article, we will go over different data providers, but for now, we will stick with SQL Server. System.Data.SqlClient is the namespace to remember if you want to connect to SQL Server.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:17]: What's a namespace

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+2. Make a command that the data object will follow.
+3. Consider the data that the data object has provided.
+
+To complete the actions listed above, we will need to employ the following classes:
+1. System.Data.SqlClient.SqlConnection
+2. System.Data.SqlClient.SqlCommand
+3. System.Data.SqlClient.SqlDataReader
+
+### Preparation
+For this project, I used a safe console application. To begin, we must declare the variables that will be used. We will need to make a variable to hold the data, one to hold the connection attributes, and another to hold the SQL statements we will be issuing. Any exceptions that we encounter when reading from the database are noted. In the working-storage section, all of these items are listed below.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:34]: Which console are you using? Are you using some kind of application? If so, mention it, and tell how to set it up.

Comment (srishilesh) [reply content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:34]: Add an image if possible

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+```SQL
+working-storage section.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:37]: Where do you find the working storage section? Add image

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+
+sqlCmd             string.
+dataFound          type System.Boolean.
+recordsRead        pic 9(06)  value zeros.
+LastName           pic x(20)  value spaces.
+FirstName          string.
+HomePhone          pic x(24)  value spaces.    
+junk               pic x(01)  value spaces.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:59]: Is this some kind of output? If so, add this under a separate code snippet

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+```
+
+It's worth noting that I can mix and match the types of data definitions. Some variables were defined using typical COBOL syntax for the PIC clause, while others were defined using standard .NET data types. By enabling this option, some of your source code can now be used without worrying about data type conversion. To read data from a table using ADO, we've generated all of the necessary divisions and references. We must now carry out the measures mentioned above.
+
+### Create a Connection
+In order to use a database, we must first establish a connection. If you've ever used ODBC with embedded SQL, you'll know that you must first establish the connection in the Control Panel's ODBC administrators panel before issuing the SQL connect query. "EXEC SQL CONNECT TO DATABASE END-EXEC" was comparable to the connect statement.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:65]: Where do you find this `Control Panel's ODBC administrators panel`?

Add image

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+The [IBM Data Server Provider for.NET](https://www.ibm.com/docs/en/db2/10.5?topic=net-provider-support-microsoft-entity-framework) must be installed for IBM. If you're using IBM DB2, you'll be dealing with IBM.Data.DB2, and if you're using Informix, you'll be working with IBM.Data.Informix.
+
+### Wrap-Up
+Data access in the Microsoft .NET environment can assist save time, automate operations, and simply show data. The steps provided in this article should assist you in comprehending the processing requirements. This article demonstrated how to develop ADO-based software by hand. Additionally, Micro Focus has provided an ADO Connection Editor and an OpenESQL utility for developing SQL-enabled source code as additional support.
+
+### Conclusion
+In this article, we have learned how to access data with ADO.NET In Micro Focus Visual COBOL. We also have learned about the basic ADO Flow of data, the Preparation, how to create a Connection, creating a Command, read the Data, understood what housekeeping is, knowing other data providers, and finally understand what Wrap-Up is.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:144]: Add other reference links for further reading

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+1. Establish a link to the data object.
+2. Make a command that the data object will follow.
+3. Consider the data that the data object has provided.
+
+To complete the actions listed above, we will need to employ the following classes:
+1. System.Data.SqlClient.SqlConnection
+2. System.Data.SqlClient.SqlCommand
+3. System.Data.SqlClient.SqlDataReader
+
+### Preparation
```
Comment (srishilesh) [inline content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:33]: What is the goal of this code tutorial? State the objectives. State the expected end result.

**Review by srishilesh — CHANGES_REQUESTED** [review]
@Amoh-writer 

Sorry for the delay in reviewing. This topic is something new to me and this looks like a more technical article, so this is taking time for me to review.

I've suggested a few a changes, please make sure to address them.

Also, please delete this file `content/articles/avatar.jpeg` since we don't use it.

Ping me when done.

Thanks.

File: content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md (hunk 4 above)
Comment (Amoh-writer) [reply content/articles/data-access-with-ado.net-in-micro-focus-visual-cobol/index.md:59]: This is not an output, we are declaring the variables in that code snippet.

**Comment by Amoh-writer** [comment]
hello @srishilesh I have completed the requested changes. Please check my last commit. 
Thank you

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
+### Introduction
+Micro Focus Visual COBOL is a modern development package that enables developers to maintain, create, and upgrade their systems for streamlining application development and delivery. Visual COBOL is a multi-language Integrated Development Environment (IDE).
+
+`ADO` is a Microsoft technology that stands for `ActiveX Data Object`. It is a collection of [ActiveX](https://en.wikipedia.org/wiki/ActiveX) controls that give you programmatic access to Microsoft's most recent data access technologies. As the name suggests, it's a wrapper for fundamental database access techniques.
+
+In simple words, ADO acts as a middleware between the programming language and the database. The developers write programs to access the data, without knowing how the database is implemented.
+
+`ADO.NET` is the next step in data access technology advancement. `ADO.NET` extends this by adding XML into a common model that applies to both relational and text-based XML data.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:8]: ``ADO.NET` extends this by adding XML into a common model that applies to both relational and text-based XML data.`

Could you explain more about this? This looks unclear

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+> Keep this namespace in mind. It's critical!
+
+There are presently 16 classes in the SQLClient namespace. These classes provide a programmatic way to establish connections to databases, verify permissions, manage exceptions, and do other duties related to data access.
+
+Let's get started using ADO.
+
+To view data object, follow this basic procedure:
+1. Establish a link to the data object.
+2. Make a command that the data object will follow.
+3. Consider the data that the data object has provided.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:75]: Could you please rephrase these 2 sentences? These sentences doesn't convey any meaning to me.

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+To complete the actions listed above, we will need to employ the following classes:
+1. System.Data.SqlClient.SqlConnection
+2. System.Data.SqlClient.SqlCommand
+3. System.Data.SqlClient.SqlDataReader
+
+### Preparation
+For this tutorial, I used a safe console application. Microsoft's Data Access technology (ADO.NET application) is one of safe console applicationt. It's a part of the `.NET` Framework that's utilized to connect the `.NET` Application to data sources.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:83]:
```suggestion
For this tutorial, I used a safe console application. Microsoft's Data Access technology (ADO.NET application) is one of safe console application. It's a part of the `.NET` Framework that's utilized to connect the `.NET` Application to data sources.
```

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+It has access to various basic data streams: standard error, standard output, and standard input.
+
+To begin, we must declare the variables that will be used. We will need to make a variable to hold the data, one to hold the connection attributes, and another to hold the SQL statements we will be issuing.
+
+Any exceptions that we encounter when reading from the database are noted.
+
+In the working-storage area, you'll find record description entries and data description entries for independent data items, referred to as data item description entries.
+
+> The working-storage section is a work ground where all the commands are written on as shon in the figure below:
+![working storage section](engineering-education/data-access-with-ado.net-in-microfocus-visual-cobol-storage-section.png)
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:94]:
```suggestion

![working storage section](engineering-education/data-access-with-ado-dotnet-in-microfocus-visual-cobol-storage-section.png)
```

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+sqlDataReader type System.Data.SqlClient.SqlDataReader.
+sqlCommand    type System.Data.SqlClient.SqlCommand.
+
+sqlCmd             string.
+dataFound          type System.Boolean.
+recordsRead        pic 9(06)  value zeros.
+LastName           pic x(20)  value spaces.
+FirstName          string.
+HomePhone          pic x(24)  value spaces.    
+junk               pic x(01)  value spaces.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:117]: I see that you've not explained about these. Please explain

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+
+### Create a Command
+Now that we've created our connection, we need to inform the database of what we're trying to accomplish. A command structure is used to do this. If you've ever used embedded SQL, you'll know what I'm talking about. Your prior statement would have been `EXEC SQL SELECT * FROM EMPLOYEES END-EXEC.` We're going to pull some data from the Employee database and present it in the console window. The following statements are used to establish the command:
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:141]: Avoid such statements: `If you've ever used embedded SQL, you'll know what I'm talking about.`

Comment (srishilesh) [reply content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:141]: Where do we have the `Employee` database?

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+
+```bash
+set sqlCmd to "SELECT FirstName, LastName, HomePhone FROM Employees".
+set sqlCommand to new type System.Data.SqlClient.SqlCommand(sqlCmd, sqlConnection).
+set sqlDataReader to sqlCommand::ExecuteReader.
+```
+
+The code above prompts the connection to only select the variables specified in the first line. Then the second line specifies the connection to be made to the `System.Data.SqlClient.SqlCommand` namespace.
+
+I established a WORKING-STORAGE variable and transferred my Select statement into it for this stage of the procedure. My user's input allows me to build a Select statement. Work can be completed in many ways, and this is just one of them. An SQL COMMAND instance must be established after the command string has been generated.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:151]: Always write from 3rd person view. Avoid using "I"

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+
+The only parameters necessary for the SQL COMMAND class to function are the connection object (you must tell it where to execute the command) and the command to execute. The `execute reader` command is the way you'll use to return the data to you after you've instantiated the SQL COMMAND. As a result, you will be provided with a data reader object that you can use to access your data. Use what method to determine if it returns data or not.
+
+### Read the Data
+OK! A database connection has been made. The data should have returned when we formed and ran a command against the connection. How can you know if any data has been returned? In addition to the alternatives listed above, here is a quick way to verify. We'll use the 'Read' method to get data from the data reader object. The Read method produces a Boolean result that indicates whether data was read or not. True or B"1" will be the value if there is information. "False" will be returned if no data has been read. DataFound will be set by adding a returning line to our read statement, and then checking if we have any data to pry from the filesystem.
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:156]: `True or B"1"`

What's B''1?

File: content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+
+### Conclusion
+In this article, we have learned how to access data with ADO.NET In Micro Focus Visual COBOL. We also have learned about the basic ADO Flow of data, the Preparation, how to create a Connection, creating a Command, read the Data, understood what housekeeping is, knowing other data providers, and finally understand what Wrap-Up is.
+
+### Further readings
+[Function to Create an ADO Connection](https://community.microfocus.com/cobol/visualcobol/f/forumid-18/348985/function-to-create-an-ado-connection)
+
+[Oracle Data Access Connectivity (ODAC) 64bit Connection issue when using ADO.Net Connection Editor](https://community.microfocus.com/cobol/visualcobol/f/forumid-18/355802/oracle-data-access-connectivity-odac-64bit-connection-issue-when-using-ado-net-connection-editor)
+
+[Database Access](https://www.microfocus.com/documentation/visual-cobol/VC231/VS2012/GUID-62A76ED2-5851-47BD-A6EC-CA6830A3C745.html)
```
Comment (srishilesh) [inline content/articles/data-access-with-ado-dotnet-in-micro-focus-visual-cobol/index.md:217]:
```suggestion
- [Function to Create an ADO Connection](https://community.microfocus.com/cobol/visualcobol/f/forumid-18/348985/function-to-create-an-ado-connection)
- [Oracle Data Access Connectivity (ODAC) 64bit Connection issue when using ADO.Net Connection Editor](https://community.microfocus.com/cobol/visualcobol/f/forumid-18/355802/oracle-data-access-connectivity-odac-64bit-connection-issue-when-using-ado-net-connection-editor)
- [Database Access](https://www.microfocus.com/documentation/visual-cobol/VC231/VS2012/GUID-62A76ED2-5851-47BD-A6EC-CA6830A3C745.html)
```

**Review by srishilesh — CHANGES_REQUESTED** [review]
@Amoh-writer 

Thanks for making the changes. This looks good.

I've suggested minor changes. Please let me know when done.

Thanks

**Comment by hectorkambow** [comment]
@srishilesh 

How many times has this PR been revised?? 

I believe this PR has been in our queue quite a while. 

If the PR comes back and still needs any revision - we will be closing it. (while the student fixes on there end via their forked repo). 

cc @Amoh-writer

**Comment by srishilesh** [comment]
> @srishilesh 
> 
> How many times has this PR been revised?? 
> 
> I believe this PR has been in our queue quite a while. 
> 
> If the PR comes back and still needs any revision - we will be closing it. (while the student fixes on there end via their forked repo). 
> 
> cc @Amoh-writer 

@hectorkambow 

This PR has been reviewed around 3 times.

I believe this would be the last round before the final revision.

Hopefully the author makes all the necessary changes that I suggested. If the author doesn't, we'll shall close the PR as you mentioned.

Thanks.

**Comment by Amoh-writer** [comment]
hello @srishilesh I have done requested changes. Thank you

**Review by srishilesh — APPROVED** [review]
@Amoh-writer 

I've made several major changes to your article.

Before starting out with your next article, I would suggest you take a very very careful look at all my changes.

Moving for the final review.

Thanks.

**Comment by ahmadmardeni1** [comment]
Welcome to Section EngEd and congrats on your first article @Amoh-writer 🎉
Please make sure to go over the changes made by your peer reviewer. This PR has been reviewed more than 3 times. We don't provide editing services and recommend that students leverage online editing services (there are both free and paid services that you can explore), or try engaging with the community or other peers in your network to have them review prior to submitting a revised proposal.

Thank you for the review @srishilesh. Great work.

**Pull request merged**